src/test/java/htsjdk/samtools/SamReaderFactoryTest.java

```diff
@@ -214,6 +217,24 @@ public void exhaustInputResourcePermutation(final SamInputResource resource) thr
         Assert.assertEquals(observedRecordOrdering.size(), 1, "read different records than other testcases");
     }
 
+    @Test
+    public void openPath() throws IOException {
+        final Path path = localBam.toPath();
+        final SamReader reader = SamReaderFactory.makeDefault().open(path);
+        LOG.info(String.format("Reading from %s ...", path));
+        final List<SAMRecord> slurped = Iterables.slurp(reader);
+        final SAMFileHeader fileHeader = reader.getFileHeader();
+        reader.hasIndex();
+        reader.indexing().hasBrowseableIndex();
+        reader.close();
+
+        /* Ensure all tests have read the same records in the same order or, if this is the first test, set it as the template. */
+        observedHeaders.add(fileHeader);
+        observedRecordOrdering.add(slurped);
+        Assert.assertEquals(observedHeaders.size(), 1, "read different headers than other testcases");
+        Assert.assertEquals(observedRecordOrdering.size(), 1, "read different records than other testcases");
```

Comment:
This test is not meaningful if it happens to run *before* `exhaustInputResourcePermutation()` has populated  `observedHeaders` and `observedRecordOrdering` instead of after it. This is a problem since TestNG by default randomizes test ordering.